pysmt/formula.py

```diff
@@ -231,74 +212,77 @@ def Times(self, left, right):
           - Left and Right must be both INT or REAL type
           - Only linear expressions are allowed
         """
-        n = self.create_node(node_type=op.TIMES, args=(left, right))
-        return n
+        return self.create_node(node_type=op.TIMES, args=(left, right))
 
+    def Pow(self, base, exponent):
+        """ Creates the n-th power of the base.
 
-    def Div(self, left, right):
-        """ Creates an expression of the form:
-            left / right
-
-        Restriction:
-          - Left and Right must be both REAL type
-          - Right is a constant
+        The exponent must be a constant.
         """
-        if not right.is_constant(types.REAL):
-            raise NonLinearError
-        inverse = Fraction(1) / right.constant_value()
-        return self.Times(left, self.Real(inverse))
+        if not exponent.is_constant():
+            raise ValueError("The exponent of POW must be a constant.", exponent)
 
+        if base.is_constant():
+            val = base.constant_value() ** exponent.constant_value()
```

Comment:
The idea here (maybe implemented incorrectly) is that constant pow is linear. By doing this simplification here, we do not need to deal with walk_pow in the converters of each solver. A solver that does not support pow, will never see the pow operator if we only have constant pow. Otherwise, we need to implement walk_pow in each converter, and check whether we are in a constant case. This issue does not exist with plus or times, because we still need to handle those operators in all LIRA converters.

Does this make sense?